imap_processing/hi/hi_l2.py

```diff
@@ -43,23 +47,34 @@ def hi_l2(
         Level 2 IMAP-Hi dataset ready to be written to a CDF file.
     """
     # TODO: parse descriptor to determine map configuration
-    sensor = "45" if "45" in descriptor else "90"
     direction: Literal["full"] = "full"
     cg_corrected = False
-    map_spacing = 4
+    map_descriptor = MapDescriptor.from_string(descriptor)
 
     rect_map = generate_hi_map(
```

Comment:
Please rename this variable to reflect that it might not be a `RectangularSkyMap`.
```suggestion
    sky_map = generate_hi_map(
```